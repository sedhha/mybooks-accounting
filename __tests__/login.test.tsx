import React from 'react';
import { render } from '@testing-library/react';
import LoginScreen from '@/components/Login';
import * as NextRouter from 'next/router';

describe('LoginScreen', () => {
	beforeEach(() => {
		// Create a Jest mock for the `useRouter` hook
		const mockRouter: NextRouter.NextRouter = {
			route: '/',
			pathname: '/',
			asPath: '/',
			basePath: '',
			isLocaleDomain: false,
			push: jest.fn(),
			replace: jest.fn(),
			reload: jest.fn(),
			back: jest.fn(),
			forward: jest.fn(),
			prefetch: jest.fn(),
			beforePopState: jest.fn(),
			events: {
				on: jest.fn(),
				off: jest.fn(),
				emit: jest.fn(),
			},
			isFallback: false,
			isReady: true,
			isPreview: false,
			query: {},
		};

		// Mock the `useRouter` hook
		jest.spyOn(NextRouter, 'useRouter').mockReturnValue(mockRouter);
	});
	afterAll(() => {
		jest.resetAllMocks();
		jest.clearAllMocks();
	});
	test('renders LoginScreen component', () => {
		const { container } = render(<LoginScreen />);
		expect(container).toMatchSnapshot();
	});
});
