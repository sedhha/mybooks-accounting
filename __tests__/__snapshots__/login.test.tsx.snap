// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen renders LoginScreen component 1`] = `
<div>
  <div
    class="screen"
  >
    <div
      class="container"
    >
      <h1>
        Login
      </h1>
      <div
        class="form"
      >
        <label
          for="login-id"
        >
          Login ID
        </label>
        <input
          id="login-id"
          name="login-id"
          required=""
          type="text"
          value=""
        />
        <div
          class="checkbox"
        >
          <label>
            User Type:
          </label>
          <div
            class="checkboxItem"
          >
            <input
              checked=""
              required=""
              type="radio"
              value="Customer"
            />
            <label>
              Customer
            </label>
          </div>
          <div
            class="checkboxItem"
          >
            <input
              required=""
              type="radio"
              value="Expert"
            />
            <label>
              Expert
            </label>
          </div>
        </div>
        <button
          type="submit"
        >
          Login
        </button>
      </div>
    </div>
  </div>
</div>
`;
